### PR TITLE
Fixed some deprecated warnings of mockito methods and minor cleanup

### PR DESCRIPTION
**Pull request by StefanSpieker**
Fixed some deprecated warnings of mockito methods and minor cleanup.
* Replaced `verifyZeroInteractions` with `verifyNoInteractions` as suggested [here](https://javadoc.io/static/org.mockito/mockito-core/3.7.0/org/mockito/Mockito.html#verifyZeroInteractions-java.lang.Object...-)
* Replaced `initMocks` with `openMocks` as suggested [here](https://www.javadoc.io/doc/org.mockito/mockito-core/3.7.0/org/mockito/MockitoAnnotations.html)

and some minor stuff, added missing `NonNull` and removed superflous `throws Exception`

### Proposed changelog entries

* Internal: N/A 

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/test/java/hudson/cli/handlers/ViewOptionHandlerTest.java
Comment (basil) [inline core/src/test/java/hudson/cli/handlers/ViewOptionHandlerTest.java:73]: As the [Mockito documentation](https://www.javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/MockitoAnnotations.html) describes, the recommended usage pattern for `MockitoAnnotations#openMocks` is to store the returned `AutoCloseable` into a `private` field in a method annotated with `@Before` and then call `Closeable#close()` in a method annotated with `@After`. Not doing this could potentially leak state, which might make it difficult to debug issues later on.

Did you consider doing this for our usages of `MockitoAnnotations`? Feel free to let me know if this is prohibitively difficult for some reason.

File: core/src/test/java/hudson/cli/handlers/ViewOptionHandlerTest.java
Comment (basil) [inline core/src/test/java/hudson/cli/handlers/ViewOptionHandlerTest.java:54]: To avoid regressing commit 819f80b12f:

```suggestion
import edu.umd.cs.findbugs.annotations.NonNull;
```

Comment (StefanSpieker) [reply core/src/test/java/hudson/cli/handlers/ViewOptionHandlerTest.java]: Oops, that was misconfigured inside my IDE, thanks for pointing out.

File: core/src/test/java/hudson/cli/handlers/ViewOptionHandlerTest.java
Comment (StefanSpieker) [reply core/src/test/java/hudson/cli/handlers/ViewOptionHandlerTest.java]: you are right, I updated the code.

**Review by basil — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
Keeping `@After` after `@Before` could have been better, but not a problem while the code is so close

**Pull request merged**
